### ISSUE TITLE
[BUG]  [INFO] No cell pixels found.

### ISSUE THREAD

**Issue by cmubioinformatics** [MouseLand/cellpose#1284]
**Describe the bug**
I am using visium HD picture, I transfer from czi to tif(extract the highest resolution layer), and export it to tif, but when I use the cellpose samto do the segmentation, show me  [INFO] No cell pixels found.

```
def process_czi_to_tif(czi_path, save_path, channels_to_use=None, channel_axis=0):
    import numpy as np
    from tifffile import imwrite
    import czifile

    # Read the CZI file
    with czifile.CziFile(czi_path) as czi:
        image = czi.asarray()

    print(f"Original CZI image shape: {image.shape}")

    # Slicing: (scene, time, channel, y, x, z)
    image = image[0, 0, :, :, :, 0]
    print("High-res shape:", image.shape)  # (2, 12096, 15039)

    image = np.transpose(image, (1, 2, 0))  # (12096, 15039, 2)
    print("Transposed shape:", image.shape)

    # Update channel_axis if needed (usually channel is now axis 0)
    # Confirm the channel axis by inspecting image.shape

    num_channels = image.shape[channel_axis]
    print(f"Number of color channels (axis {channel_axis}): {num_channels}")

    if channels_to_use is not None:
        print(f"Selected channel indices to merge: {channels_to_use}")
        channels = [np.take(image, i, axis=channel_axis) for i in channels_to_use]
    else:
        channels = [np.take(image, i, axis=channel_axis) for i in range(num_channels)]
        print(f"No channels specified, merging all channels: {list(range(num_channels))}")

    # Merge channels, using axis=0 as the channel dimension for TIFF
    merged = np.stack(channels, axis=0)
    print(f"Number of color channels in merged TIFF: {merged.shape[0]}")

    # Save as multi-channel TIFF
    imwrite(save_path, merged)
    print(f"Multi-channel TIFF saved to: {save_path}")


def show_tiff_channels(image_path):
    img = tifffile.imread(image_path)
    print("Image shape:", img.shape)
    if img.ndim == 2:
        print("This is a grayscale image with no color channels.")
        plt.imshow(img, cmap='gray')
        plt.title("Grayscale Image")
        plt.axis('off')
        plt.show()
    elif img.ndim == 3:
        if img.shape[0] <= 10:
            channel_axis = 0
        elif img.shape[2] <= 10:
            channel_axis = 2
        else:
            print("Cannot determine the channel axis. Please check your data.")
            return
        num_channels = img.shape[channel_axis]
        print(f"This is a multi-channel image with {num_channels} channels (channel axis: {channel_axis}).")
        plt.figure(figsize=(10, 5))
        for i in range(min(2, num_channels)):
            plt.subplot(1, 2, i+1)
            if channel_axis == 0:
                channel = img[i]
            else:
                channel = img[:, :, i]
            plt.imshow(channel, cmap='gray')
            plt.title(f"Channel {i} (gray)")
            plt.axis('off')
        plt.show()
    else:
        print("Image has more than 3 dimensions. This may be a complex multi-dimensional dataset.")

```

```
czi_path = '../0_alignment/0_czi_image/L2_4_STITCHED.czi'  # original czi image
save_path = './tif/L2_nm.tif'  # save path for the merged tif image


process_czi_to_tif(
    czi_path= czi_path,
    save_path = save_path , 
    channels_to_use=[0, 1],
    channel_axis=2  
)

```
Here is the logs

```
Original CZI image shape: (1, 1, 2, 12096, 17496, 1)
High-res shape: (2, 12096, 17496)
Transposed shape: (12096, 17496, 2)
Number of color channels (axis 2): 2
Selected channel indices to merge: [0, 1]
Number of color channels in merged TIFF: 2
Multi-channel TIFF saved to: ./tif/L2_nm.tif
```

It looks like transferring successfully

and then I check two color channels
```
show_tiff_channels(save_path)
```

<img width="829" height="314" alt="Image" src="https://github.com/user-attachments/assets/155397d3-7ee5-497e-864d-53f0ecbb5117" />

looks good, this tif file contains color channels

and then, I import this image to cellpose
```
import numpy as np
import tifffile
import matplotlib.pyplot as plt
from cellpose import models, core, io
import os
import numpy as np
from skimage import io  # Assuming io.imread is from skimage
```
```
Welcome to CellposeSAM, cellpose v
cellpose version: 	4.0.6 
platform:       	linux 
python version: 	3.8.5 
torch version:  	2.4.1+cu121! The neural network component of
CPSAM is much larger than in previous versions and CPU excution is slow. 
We encourage users to use GPU/MPS if available. 
```
good, with gpu

read in the transferred tif file
```
dir = "./tif/"
dir = Path(dir)
if not dir.exists():
  raise FileNotFoundError("directory does not exist")

# *** change to your image extension ***
image_ext = "L2_nm.tif"
# L2_4_STITCHED_nm
# list all files
files = natsorted([f for f in dir.glob("*"+image_ext) if "_masks" not in f.name and "_flows" not in f.name])

if(len(files)==0):
  raise FileNotFoundError("no image files found, did you specify the correct folder and extension?")
else:
  print(f"{len(files)} images in folder:")

for f in files:
  print(f.name)
```
here is the logs
```
1 images in folder:
L2_nm.tif
```
file exists

then check the shape
```
img = io.imread(files[0])
print(f"Original shape: {img.shape}")
```

```
Original shape: (2, 12096, 17496)
```
looks good

then, according official website to extract  the color channels to new object(I think this is not necessary for me here, but respect)
```
first_channel = '0' # @param ['None', 0, 1, 2, 3, 4, 5]
second_channel = '1' # @param ['None', 0, 1, 2, 3, 4, 5]
third_channel = 'None' # @param ['None', 0, 1, 2, 3, 4, 5]

selected_channels = []
for i, c in enumerate([first_channel, second_channel, third_channel]):
  if c == 'None':
    continue
  if int(c) > img.shape[-1]:
    assert False, 'invalid channel index, must have index greater or equal to the number of channels'
  if c != 'None':
    selected_channels.append(int(c))

img_selected_channels = np.zeros_like(img)
img_selected_channels[:, :, :len(selected_channels)] = img[:, :, selected_channels]
```

and then, cellpose

```
flow_threshold = 0.4
cellprob_threshold = 0.0
tile_norm_blocksize = 200

masks, flows, styles = model.eval(img_selected_channels, 
                                  batch_size=32, 
                                  flow_threshold=flow_threshold, 
                                  cellprob_threshold=cellprob_threshold)
# ,normalize={"tile_norm_blocksize": tile_norm_blocksize}

```

<img width="648" height="239" alt="Image" src="https://github.com/user-attachments/assets/ea4765b5-a52b-4467-833a-1ceb8cfcebb7" />

```
2025-07-26 15:21:04,051 [INFO] No cell pixels found.
```

then I also check the cellpose output mask file

<img width="722" height="536" alt="Image" src="https://github.com/user-attachments/assets/7654fa81-0419-41dd-b1b1-79c1ca3ae073" />

really weird

I don't know what's wrong, somebody can give me some advice?

I know we can use FIJI to transfer czi to tif file, but I want to process this in python.

Thanks very much!!

**Comment by cmubioinformatics** [comment]
It is possible the transferring problem, but I  double check, the transferred tif file contains color channels, shape is also right. make me confused. no idea what's the problem.

**Comment by mrariden** [comment]
Can you trouble shoot with a smaller image that you get off the internet? Maybe [this one](https://www.cellpose.org/static/images/img00.png)? Cellpose should work out of the box if you use our code. 

I'd check that your `img_selected_channels` variable has the correct data in it. 

Otherwise, you will get more help at the image.sc forum if you don't have any library errors.
